### PR TITLE
modules/aws/ignition/resources/s3-puller.sh: add retries to avoid races

### LINKED ISSUE #924: S3 download failure causes cluster bootstrap failure
If the init-assets unit fails (due to an S3 download error), it causes a cascading failure to the other dependent units: bootkube & tectonic.

#796 documents the issue well, but it's unclear if that is a complete fix.

### PR DESCRIPTION
**Pull request by Quentin-M**
Fixes #924

### CODE REVIEW

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
Me neither. Let me remove it..

**Comment by Quentin-M** [comment]
ok to test

File: modules/aws/ignition/resources/s3-puller.sh [hunk 1]
```diff
@@ -12,7 +12,10 @@ fi
     --set-env="LOCATION=$1" \
     --exec=/bin/bash -- -c '
         REGION=$(wget -q -O - http://169.254.169.254/latest/meta-data/placement/availability-zone | sed s'/[a-zA-Z]$//')
-        /usr/bin/aws --region=$${REGION} s3 cp s3://$${LOCATION} /tmp/$${LOCATION//\//+}
+        until /usr/bin/aws --region=$${REGION} s3 cp s3://$${LOCATION} /tmp/$${LOCATION//\//+}; do
+          echo "Could not pull from S3, retrying in 5 seconds ($i)"
```
Comment (s-urbaniak) [inline modules/aws/ignition/resources/s3-puller.sh]: post-merge nit: I believe `$i` will stay 0 in the logs as it is unused/undeclared.

Comment (Quentin-M) [inline modules/aws/ignition/resources/s3-puller.sh]: Woopsy! Thanks :D

Comment (Quentin-M) [inline modules/aws/ignition/resources/s3-puller.sh]: https://github.com/coreos/tectonic-installer/pull/1089

**Pull request merged**